js/command.ts

```diff
@@ -0,0 +1,193 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as dispatch from "./dispatch";
+import { DenoError, ErrorKind } from "./errors";
+import * as flatbuffers from "./flatbuffers";
+import * as msg from "gen/msg_generated";
+import { assert, isObject, unreachable } from "./util";
+
+export interface CommandOptions {
+  argv?: string[];
+  dir?: string | null;
```

Comment:
If we were to expose the lowest common denominator API of starting a process, it wouldn't have this 'dir' option. It seems python's `subprocess.Popen` does not have such an option. Presumably its implemented by doing fork-chdir-exec. One could argue the chdir is the responsibility of the subprocess.

Reply:
> If we were to expose the lowest common denominator API of starting a process, it wouldn't have this 'dir' option.
> It seems python's subprocess.Popen does not have such an option

The lowest common denominator of what? of programming languages? I'm pretty sure brainfuck doesn't support it. 

But every serious one does. Python's Popen calls it 'cwd'; so do php, nodejs and haskell. Rust uses builder object with a `workingDirectory`, method; java uses a method too but they just call it `directory`. C# on the other hand does not use a builder object but they also call it `WorkingDirectory` just like Java but with different capitalization. Ruby calls it `chdir` which IMO is quite weird. Dart calls it workingDirectory like Java. Boost calls it start_dir. 

The name `dir` comes from go.

> One could argue the chdir is the responsibility of the subprocess.

Give it a spin.

Reply:
GoLang feels like the outlier, but with the wonders of JSDoc, intent can be clear no matter what the name...

I know I would personally expect the ability to specify the cwd when raising the process... I would really hate to have to write a shell script and a command file, detect which OS, and switch just so I could execute a "dumb" binary that is being pedantic about what its cwd is.

Reply:
@kitsonk Agreed; switched to `cwd`, that seems the most common.
I picked `dir` because `cwd` is a little weird to capitalize -- does one do `Cwd` or `CWD` ?
But I realize that it's all internal flatbuffer code and nothing about our api, so it doesn't matter.